conf/topics.json

```diff
@@ -1,122 +1,119 @@
-                    "Killed"
-                ],
-                "type": "string"
+                "type": "string",
+                "description": "Standardized source application name (aqueduct, unify, lum, etc)"
             },
-            "timestamp_end": {
-                "description": "End timestamp of the run in epoch milliseconds",
-                "type": "number"
+            "source_app_version": {
+                "type": "string",
+                "description": "Source application version (SemVer preferred)"
+            },
+            "environment": {
+                "type": "string",
+                "description": "Environment (dev, uat, pre-prod, prod, test or others)"
             },
             "timestamp_start": {
-                "description": "Start timestamp of the run in epoch milliseconds",
-                "type": "number"
+                "type": "number",
+                "description": "Start timestamp of the run in epoch milliseconds"
+            },
+            "timestamp_end": {
+                "type": "number",
+                "description": "End timestamp of the run in epoch milliseconds"
+            },
+            "jobs": {
+                "type": "array",
+                "description": "List of individual jobs withing the run",
+                "element_type": "object",
+                "object_schema": {
+                    "catalog_id": {
+                        "type": "string",
+                        "description": "Identifier for the data definition (Glue/Hive) database and table name for  example"
+                    },
+                    "status": {
+                        "type": "string",
+                        "enum": ["succeeded", "failed", "killed", "skipped"],
+                        "description": "Status of the job."
+                    },
+                    "timestamp_start": {
+                        "type": "number",
+                        "description": "Start timestamp of a job that is a part of a run in epoch milliseconds"
+                    },
+                    "timestamp_end": {
+                        "type": "number",
+                        "description": "End timestamp of a job that is a part of a run in epoch milliseconds"
+                    },
+                    "message": {
+                        "type": "string",
+                        "description": "Job status/error message."
+                    }
+                }
             }
         },
-        "required": [
-            "guid",
-            "app_id_snow",
-            "source_app",
-            "timestamp_start",
-            "timestamp_end",
-            "data_definition_id",
-            "status"
-        ],
-        "type": "object"
+        "required": ["event_id", "job_ref", "tenant_id", "source_app", "source_app_version", "environment", "timestamp_start", "timestamp_end", "jobs", "jobs.catalog_id", "jobs.status", " jobs.timestamp_start", "jobs.timestamp_end"]
```

Comment:
Not 100% sure how it works when `jobs` is a mandatory field. It is an array, and it is possible to have an empty array when the pipeline has failed and was not able to run any jobs. For instance, when EMR is not available 😆

Reply:
actually.... we need to remove
jobs.*   required fields... as jsonschema validate complains as that prescribes them to be present in underlying object, not in the element of a array

those are
jobs.catalog_id
jobs.status
jobs.timestamp_start
jobs.timestamp_end
as none of those would be present at that xpath

as for jobs..... schema just requires field to be there and to be array.... so empty array is fine.... null is not fine though